client/src/components/MainPage/TaskModal/TagInput.tsx

```diff
@@ -0,0 +1,198 @@
+import React, { useState, useEffect, useRef } from 'react';
+import axios from 'axios';
+import { HOST } from '../../../constants';
+import styled from 'styled-components';
+import useInput from '../../../hooks/useInput';
+import { RiCloseLine } from 'react-icons/ri';
+import { Tag } from 'GlobalType';
+
+axios.defaults.withCredentials = true; // withCredentials 전역 설정
+
+interface Props {
+  setTagIdx: React.Dispatch<React.SetStateAction<number | undefined>>;
+}
+
+const TagInput = (props: Props) => {
+  const [tagList, setTagList] = useState<Tag[]>([]);
+  const [tagInput, onTagInputChange, setTagInput] = useInput('');
+  const [isTagFocus, setIsTagFocus] = useState<Boolean>(false); // Input창이 Focus될 때만 List 표시
+  const [selectedTag, setSelectedTag] = useState<Tag>(); // 선택된 하나의 태그
+  const [randomColor, setRandomColor] = useState(''); // 새로운 Tag 추가 시 색깔 랜덤 배정
```

Comment:
해당 상태값에 색이 배정될 때 사용되는 로직이 `random`일 뿐,
변수가 의미하는 바 자체는 `새 태그에 입혀질 색깔`인 것 같습니다.
```suggestion
  const [newTagColor, setNewTagColor] = useState(''); // 새로운 Tag 추가 시 색깔 랜덤 배정
```